src/components/studentForm/AddStudent.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Select from 'react-select';
+import { useForm, Controller } from 'react-hook-form';
+import { Input, Button } from '@mui/material';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { schema } from './FormValidation';
+import GradeData from '../../utils/data/GradeData';
+import SubjectData from '../../utils/data/SubjectData';
+import * as yup from 'yup';
+import { IStudent } from '../../state/ducks/student/types';
+
+import './formStyle.css';
+import { useDispatch } from 'react-redux';
+import moment from 'moment';
+
+type IProps = {
+  AddStudentData: (data: IStudent) => void;
+};
+
+const AddStudent = ({ AddStudentData }: IProps) => {
+  const { control, handleSubmit } = useForm();
+
+  const onSubmit = (data: any) => {
+    let student = {
+      name: data.name,
+      marks: data.marks,
+      subject: data.subject.value,
+      grade: data.grads.value,
+      date: moment().toISOString(),
+    };
+    return AddStudentData(student);
+  };
```

Comment:
```suggestion
  const onSubmit = (data: IStudent) => {
    let student = {
      ...data,
      subject: data.subject.value,
      grade: data.grads.value,
      date: moment().toISOString(),
    };
    return AddStudentData(student);
  };
```
Try to refactor your code once you have written it.